services/subscription.go

```diff
@@ -156,7 +160,9 @@ func (sub RPCLogSubscription) listenToSubscriptionErrors() {
 func (sub RPCLogSubscription) listenToLogs() {
 	sub.backfillWG.Wait()
 	for el := range sub.logs {
-		sub.dispatchLog(el)
+		if _, present := sub.backfilledSet[el.BlockHash.String()]; !present {
+			sub.dispatchLog(el)
```

Comment:
An open question here is what to use to key off a backfilled log: `BlockNumber` or `BlockHash`. This only comes into play in the scenario of a fork, which we've discussed and opted to tackle later as an epic. So for now, I believe it's a wash. I opted for `BlockHash` because I felt it was more unique, but can always change to `BlockNumber` if you prefer @se3000.